.travis.yml

```diff
@@ -32,17 +32,18 @@ before_install:
 install:
   - conda create -n testenv pip python=$PYTHON
   - source activate testenv
-  - cat ci/deps_${DEPS}.txt ci/utils.txt > deps.txt
+  # Conda's matplotlib package pulls qt but this is not required for
+  # non-interactive tests; use the compact matplotlib-base instead
+  - if [ $MPLBACKEND == "qt5agg" ];
+    then cat ci/deps_${DEPS}.txt ci/utils.txt > deps.txt;
+    else cat ci/deps_${DEPS}.txt ci/utils.txt | sed 's/matplotlib$/matplotlib-base/' > deps.txt;
+    fi
   - conda install --file deps.txt
   - pip install .
 
 
 before_script:
-  - if [ $MPLBACKEND == "qtagg" ]; then
-    export DISPLAY=:99.0;
-    sh -e /etc/init.d/xvfb start;
-    sleep 3;
-    fi
+  - python ci/cache_test_datasets.py
```

Comment:
I may be missing something but how does this line replace the xvfb setup?

Reply:
It is declared earlier as a service:
https://github.com/mwaskom/seaborn/blob/eb690eca5ced20de231c0bdc8ed5901b7e0e5f83/.travis.yml#L5-L6

Reply:
Oh cool.